src/components/Navbar/SettingsDrawer/TafsirSection.tsx

```diff
@@ -16,18 +18,68 @@ import {
   increaseTafsirFontScale,
   decreaseTafsirFontScale,
 } from 'src/redux/slices/QuranReader/styles';
+import { selectSelectedTafsirs } from 'src/redux/slices/QuranReader/tafsirs';
 import QuranReaderStyles from 'src/redux/types/QuranReaderStyles';
+import { makeTafsirsUrl } from 'src/utils/apiPaths';
+import { areArraysEqual } from 'src/utils/array';
+import { TafsirsResponse } from 'types/ApiResponses';
 
 const TafsirSection = () => {
-  const { t } = useTranslation('common');
+  const { t, lang } = useTranslation('common');
   const dispatch = useDispatch();
   const quranReaderStyles = useSelector(selectQuranReaderStyles, shallowEqual) as QuranReaderStyles;
   const { tafsirFontScale } = quranReaderStyles;
+  const selectedTafsirs = useSelector(selectSelectedTafsirs, areArraysEqual);
+
+  const tafsirLoading = useCallback(
+    () => (
+      <div>
+        {selectedTafsirs.map((id) => (
+          <Skeleton key={id}>
+            <div>{id}</div>
+          </Skeleton>
+        ))}
+      </div>
+    ),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [selectedTafsirs.length],
+  );
+
+  const renderTafsirs = useCallback(
+    (data: TafsirsResponse) => {
+      const firstSelectedTafsir = data.tafsirs.find((tafsir) => tafsir.id === selectedTafsirs[0]);
+
+      let selectedValueString = t('settings.no-tafsir-selected');
+      if (selectedTafsirs.length === 1) selectedValueString = firstSelectedTafsir.name;
+      if (selectedTafsirs.length > 1)
+        selectedValueString = t('settings.value-and-others', {
+          value: firstSelectedTafsir.name,
+          othersCount: selectedTafsirs.length - 1,
+        });
+
+      return (
+        <SelectionCard
+          label="Selected Translations"
+          value={selectedValueString}
+          onClick={() => dispatch(setSettingsView(SettingsView.Tafsir))}
+        />
+      );
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [selectedTafsirs.length],
```

Comment:
`selectedTafsirs` should be the dependency instead of `selectedTafsirs.length` because we are accessing it inside this code `selectedTafsirs[0]`